### PR TITLE
Allow transmute() with NULL input to run without error

### LINKED ISSUE #163: Can't create new column via transmute() with NULL as input expression

### PR DESCRIPTION
**Pull request by thisisnic**
Fixes #163

### CODE REVIEW

File: R/compiler.R [hunk 1]
```diff
@@ -137,6 +137,12 @@ SubstraitCompiler <- R6::R6Class(
     #'
     #' @return `self`
     validate = function() {
+
+      # DuckDB backend doesn't accept empty SELECT clause
+      if (inherits(self, "DuckDBSubstraitCompiler") && length(self$schema$names) == 0) {
+        rlang::abort("Column list must not be empty")
+      }
```
Comment (paleolimbot) [inline R/compiler.R:144]: I think this needs to go in pkg-duckdb.R since it's specific to the `DuckDBSubstraitCompiler`. You can redefine the validate method like this: `validate = function() { super$validate(); ...duckdb-specific stuff...; self }` (in the `DuckDBSubstraitCompiler` R6 class next to all the other methods).

(Also, you should add a test in test-pkg-duckdb.R that just tests this bit of code. Off the top of my head, `substrait_project()` is a more minimal way to test this error message in that file).

File: tests/testthat/test-dplyr-mutate.R [hunk 2]
```diff
@@ -92,13 +92,23 @@ test_that("transmute respect bespoke dplyr implementation", {
 })
 
 test_that("transmute() with NULL inputs", {
```
Comment (paleolimbot) [inline tests/testthat/test-dplyr-mutate.R:94]:
```suggestion
test_that("transmute() with only NULL inputs", {
```

(unless this is also testing other kinds of transmute-with-null inputs, which might be a good idea since if there area only null inputs than it can't be tested on DuckDB)

**Review by paleolimbot — COMMENTED** [review]
Very close! The logic is sound but the validate method should be moved to the duckdb-specific compier and a test with a narrower scope should be added in test-pkg-duckdb.R.

File: R/compiler.R (hunk 1 above)
Comment (thisisnic) [reply R/compiler.R]: Ah yeah, of course it belongs there; cheers!  

Out of interest, why does it need `super$validate()`? Is that in case we ever put other stuff in the parent class that we want to do before we call the DuckDB stuff?

And, done!

File: tests/testthat/test-dplyr-mutate.R (hunk 2 above)
Comment (thisisnic) [reply tests/testthat/test-dplyr-mutate.R:94]: > other kinds of transmute-with-null inputs

Sorry, I don't follow, how do you mean?

Comment (paleolimbot) [reply tests/testthat/test-dplyr-mutate.R:94]: I was thinking `transmute(int = NULL, something_else = dbl)`?

Comment (thisisnic) [reply tests/testthat/test-dplyr-mutate.R:94]: OK, I won't merge your suggestion but have added in another test as I figure this is something that changing the implementation could break it, and no harm in more tests right?

**Review by paleolimbot — APPROVED** [review]

File: R/compiler.R (hunk 1 above)
Comment (paleolimbot) [reply R/compiler.R]: That's the idea! I don't remember if we do anything important there now but it seemed like a good idea?

Comment (thisisnic) [reply R/compiler.R]: Nothing now, but cool, I see why we would!
